pkg/syncer/cnsoperator/controller/csinodetopology/csinodetopology_controller.go

```diff
@@ -275,6 +323,95 @@ func (r *ReconcileCSINodeTopology) Reconcile(ctx context.Context, request reconc
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Infof("CSINodeTopology resource with name %q not found. Ignoring since object must have "+
+				"been deleted.", request.Name)
+			return reconcile.Result{}, nil
+		}
+		log.Errorf("Failed to fetch the CSINodeTopology instance with name: %q. Error: %+v", request.Name, err)
+		// Error reading the object - return with err.
+		return reconcile.Result{}, err
+	}
+
+	// Initialize backOffDuration for the instance, if required.
+	var timeout time.Duration
+	func() {
+		backOffDurationMapMutex.Lock()
+		defer backOffDurationMapMutex.Unlock()
+		if _, exists := backOffDuration[instance.Name]; !exists {
+			backOffDuration[instance.Name] = time.Second
+		}
+		timeout = backOffDuration[instance.Name]
+	}()
+
+	// Fetch topology labels for guest worker node backed by vmop VM.
+	topologyLabels, err := getNodeTopologyInfoForGuest(ctx, instance, r.vmOperatorClient, r.supervisorNamespace)
+	if err != nil {
+		msg := fmt.Sprintf("failed to fetch topology information for the worker node %q. Error: %v",
+			instance.Name, err)
+		log.Error(msg)
+		_ = updateCRStatus(ctx, r, instance, csinodetopologyv1alpha1.CSINodeTopologyError, msg)
+		return reconcile.Result{RequeueAfter: timeout}, nil
+	}
+
+	// Update CSINodeTopology instance.
+	instance.Status.TopologyLabels = topologyLabels
+	if err := updateCRStatus(ctx, r, instance, csinodetopologyv1alpha1.CSINodeTopologySuccess,
+		fmt.Sprintf("Topology labels successfully updated for the worker node %q", instance.Name)); err != nil {
+		return reconcile.Result{RequeueAfter: timeout}, nil
+	}
+
+	// On successful event, remove instance from backOffDuration.
+	func() {
+		backOffDurationMapMutex.Lock()
+		defer backOffDurationMapMutex.Unlock()
+		delete(backOffDuration, instance.Name)
+	}()
+
+	log.Infof("Successfully updated topology labels for worker %q in %s",
+		instance.Name, cnstypes.CnsClusterFlavorGuest)
+	return reconcile.Result{}, nil
+}
+
+func getNodeTopologyInfoForGuest(ctx context.Context, instance *csinodetopologyv1alpha1.CSINodeTopology,
+	vmOperatorClient client.Client, supervisorNamespace string) ([]csinodetopologyv1alpha1.TopologyLabel, error) {
+	log := logger.GetLogger(ctx)
+
+	virtualMachine := &vmoperatortypes.VirtualMachine{}
+	vmKey := types.NamespacedName{
+		Namespace: supervisorNamespace,
+		Name:      instance.Name, // use the nodeName as the VM key
+	}
+
+	var err error
+	if err = vmOperatorClient.Get(ctx, vmKey, virtualMachine); err != nil {
+		return nil, logger.LogNewErrorf(log,
+			"failed to get VirtualMachines for the node: %q. Error: %+v", instance.Name, err)
+	}
+
+	var topologyLabels []csinodetopologyv1alpha1.TopologyLabel
+	if virtualMachine.Status.Zone != "" {
+		topologyLabels = make([]csinodetopologyv1alpha1.TopologyLabel, 0)
+		topologyLabels = append(topologyLabels,
+			csinodetopologyv1alpha1.TopologyLabel{
+				Key:   common.WellKnownTopologyLabelsDomain + "/" + common.TopologyLabelsZoneKey,
```

Comment:
Can you use the `LabelTopologyZone` constant from "k8s.io/api/core/v1" library here? If yes, can you remove the constants you have defined in the beginning of the PR too?

Reply:
Good suggestion, I will check the constants.